metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistration.java

```diff
@@ -118,6 +124,13 @@ public Builder setIsMigratingZkBroker(boolean isMigratingZkBroker) {
             return this;
         }
 
+        public Builder setDirectories(Collection<Uuid> directories) {
```

Comment:
I think it would be better to just unconditionally copy this. It's a bit of a tricky pattern to sometimes copy and sometimes not (doesn't play well with mutability). Since this isn't a performance hotspot, it would be good to avoid.